libsam.c

```diff
@@ -839,6 +842,178 @@ int sam_put_user_grpids(struct hive *hdesc, int rid, struct keyval *val)
 
 /********* GROUP / USER MANIPULATION ROUTINES **************/
 
+/* Add SID to a group
+ * SID = any SID
+ * grp = group ID
+ * return true if success
+ */
+
+int sam_add_sid_to_grp(struct hive *hdesc, struct sid_binary * sid, int grp)
+{
+    struct sid_array *sarray, *narray;
+    struct sid_binary *usid = sid;
+    int members, newmembers;
+    char *str;
+    int o, n, hit, c;
+
+    if (!sid || !grp || (hdesc->type !=HTYPE_SAM) ) return(0);
+
+    if (gverbose)
+    {
+        str = sam_sid_to_string(usid);
+        printf("sam_add_sid_to_grp: user SID is <%s>\n", str);
+        free(str);
+    }
+
```

Comment:
gverbose should only be used to turn on or off the printf statement.
Else if gverbose is false then your code does not do anything useful.